sips/fork_support.md

```diff
@@ -20,56 +20,260 @@ Client implementation should calculate when a new duty starts to decide if it st
+func (networkID NetworkID) ForkAtEpoch(epoch phase0.Epoch) (*ForkData, error) {
+	// Get list of forks
+	forks := networkID.GetForksData()
+
+	// If empty, raise error
+	if len(forks) == 0 {
+		return nil, errors.New("GetCurrentFork: fork list by GetForksData is empty.")
+	}
+
+	var current_fork *ForkData
+	for _, fork := range forks {
+		if fork.Epoch <= epoch {
+			current_fork = fork
+		}
+	}
+	return current_fork, nil
+}
+
+func (f ForkData) GetRoot() ([]byte, error) {
+	byts, err := json.Marshal(f)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal ForkData")
+	}
+	ret := sha256.Sum256(byts)
+	return ret[:], nil
+}
+```
+
+### Structures access to types
+
+- `Share`: It's more correct for the structure to hold the information on the network that it's in, instead of the network and fork number.
+    Thus, _Share_ will contain a `NetworkID` field instead of a `DomainType` field.
+- `Config`: Similarly, _Config_ should have a `NetworkID` instead of `DomainType`.
+- `Controller`: It may drop its `DomainType` field since it has access to `config` and `Share`.
+
+Regarding the indirect usage of Identifiers:
+- `Instance` must keep its identifier to send properly formed messages.
+- `Controller` may keep its identifier, but its _StartNewInstance_ method should receive as an argument the new identifier for the instance, as well as the new _DomainType_ to update its _config_. The controller, then, updates its identifier.
+
+```go
+func (c *Controller) StartNewInstance(height Height, value []byte, identifier []byte, domainType types.DomainType) error { // <-- identifier as argument
+	if err := c.GetConfig().GetValueCheckF()(value); err != nil {
+		return errors.Wrap(err, "value invalid")
+	}
+
+	// can't use <= because of height == 0 case
+	if height < c.Height {
+		return errors.New("attempting to start an instance with a past height")
+	}
+
+	// covers height == 0 case
+	if c.StoredInstances.FindInstance(height) != nil {
+		return errors.New("instance already running")
+	}
+
+    c.config.SetSignatureDomainType(domainType) // <-- updates its config's DomainType
+    c.Identifier = identifier // <-- updates its identifier
+	c.Height = height
+	newInstance := c.addAndStoreNewInstance()
+	newInstance.Start(value, height)
+
+	c.forceStopAllInstanceExceptCurrent()
+
+	return nil
+}
 ```
 
+The above change also requires an update in the `BaseRunner`. In the `decide` function, where `StartNewInstance` is called, an updated identifier needs to be passed as an argument as well as the new DomainType. For that, the _BaseRunner_ will have a new attribute `CurrentFork`, updated every time a new duty starts.
+
+```go
+type BaseRunner struct {
+	State          *State
+	Share          *types.Share
+	QBFTController *qbft.Controller
+	BeaconNetwork  types.BeaconNetwork
+	BeaconRoleType types.BeaconRole
+	highestDecidedSlot spec.Slot
+
+	// Current fork associated to Share.DomainType
+	CurrentFork *types.ForkData  // <-- new field
```

Comment:
We can get form from domain right? why add it?

Reply:
Yes, we can. Updating :)